### PR TITLE
Fix images in Notebook not showing

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/18062

The issue here is that VS Code changed it so that all direct file:// access is blocked - images have to be loaded using the vscode-file protocol now. So the fix here is to use `asBrowserUri` to transform the URI into one that we can load when the URI is a file on disk.

See https://github.com/microsoft/vscode/issues/80352 for a similiar-ish issue in their own markdown renderer

Notebook with local, web and embedded images : 
image

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts [hunk 1]
```diff
@@ -98,7 +99,14 @@ export class NotebookMarkdownRenderer {
 			}
 			let attributes: string[] = [];
 			if (href) {
-				attributes.push(`src="${href}"`);
+				// VS Code blocks loading directly from the file protocol - we have to transform it to a vscode-file URI
+				// first. Since the href here can be either a file path, an HTTP/S link or embedded data though we first
+				// check if it's any of the others and if so then don't need to do anything.
+				let uri = URI.parse(href);
+				if (!(uri.scheme === 'https' || uri.scheme === 'http' || uri.scheme === 'data')) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts:106]: This isn't my favorite thing to do but I couldn't think of a better way to determine if a random string is a file path or something else....open to any other suggestions though!

Comment (chlafreniere) [reply src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts:106]: What happens for relative paths that are on-disk? `"./foo.png"` Does the scheme get set properly here?

Comment (chlafreniere) [reply src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts:106]: Oh never mind, I see now, this is only for non-file schemes. Carry on 😄

**Comment by chlafreniere** [comment]
@Charles-Gagnon please fix the tests 😄

**Comment by barbaravaldez** [comment]
@Charles-Gagnon I manually tested your fix and I found an issue were the `vscode-file://vscode-app/` scheme is added twice to the image from disk uri after editing the text cell in WYSIWYG mode.

Here are the repro steps: 
1. Insert image from disk (not an attachment)
2. Change to WYSIWYG mode and edit cell
3. Exit cell
Screen Shot 2022-01-25 at 8 39 19 PM

**Comment by Charles-Gagnon** [comment]
Ohhhh, yeah whoops. Good catch @barbaravaldez !

**Comment by Charles-Gagnon** [comment]
Overall I'm not thrilled about this fix, but I don't see any other way to handle it given how we render Notebooks. VS Code has stopped rendering images from the local filesystem (except for stuff in the app/extension folders) and as far as I can tell the only other solutions are : 

1. Revert app protocol changes to allow loading content directly in renderer process from anywhere on the filesystem
2. Change our Notebook MD rendering to be hosted in its own webview, similar to how the MD extension does it

Neither of which really seems feasible at this point.

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**